scripts/ami-helper/ami-helper.sh

```diff
@@ -0,0 +1,193 @@
+	SECRET_NAME="peer-pods-image-creation-secret"
+	echo "Temporary Workdir: ${TMPDIR}"
+}
+
+init() {
+	command -v oc &> /dev/null ||  { echo "oc command was not found" 1>&2 ; exit 1; }
+	oc cluster-info &> /dev/null ||  { echo "cluster is not configured" 1>&2 ; exit 1; }
+	[[ -x "$(command -v aws)" ]] || { echo "aws is not installed" 1>&2 ; exit 1; }
+	aws sts get-caller-identity &>/dev/null || { echo "aws cli missing credentials"; exit 1; }
+	[[ $REGION ]] || REGION=$(oc get infrastructure -n cluster -o=jsonpath='{.items[0].status.platformStatus.aws.region}') || { echo "Region couln't be fetched, add as argument" 1>&2 ; exit 1; }
+	[[ ! $BUCKET_NAME ]] && uid=$(kubectl get infrastructure -n cluster -o jsonpath='{.items[*].metadata.uid}') && BUCKET_NAME=osc-${uid:0:6}-bucket && \
+	echo "Bucket name not provided, using ${BUCKET_NAME}, make sure to set it in the aws-podvm-image-cm"
+
+	echo "Bucket Name: ${BUCKET_NAME}"
+	echo "Region: ${REGION}"
+}
+
+delete_bucket() {
+	echo "Delete s3 Bucket named ${BUCKET_NAME} at ${REGION}"
+	aws s3api delete-bucket --bucket ${BUCKET_NAME} --region ${REGION}
+}
+
+create_bucket() {
+	echo "Create s3 Bucket named ${BUCKET_NAME} at ${REGION}"
+	if [[ ${REGION} == us-east-1 ]]; then
+		aws s3api create-bucket  --bucket ${BUCKET_NAME} --region ${REGION}
+	else
+		aws s3api create-bucket  --bucket ${BUCKET_NAME} --region ${REGION} --create-bucket-configuration LocationConstraint=${REGION}
+	fi
+}
+
+set_service_role() {
+	echo "Create the service role"
+	cat <<EOF > "${TRUST_POLICY_JSON_FILE}"
+{
+	"Version":"2012-10-17",
+	"Statement":[
+		{
+			"Effect":"Allow",
+			"Principal":{ "Service":"vmie.amazonaws.com" },
+			"Action": "sts:AssumeRole",
+			"Condition":{"StringEquals":{"sts:Externalid":"vmimport"}}
+		}
+	]
+}
+EOF
+
+	aws iam create-role --role-name vmimport --assume-role-policy-document "file://${TRUST_POLICY_JSON_FILE}" --region ${REGION}
+
+	echo "Attach policy"
+	cat <<EOF > "${ROLE_POLICY_JSON_FILE}"
+{
+	"Version":"2012-10-17",
+	"Statement":[
+		{
+			"Effect":"Allow",
+			"Action":["s3:GetBucketLocation","s3:GetObject","s3:ListBucket"],
+			"Resource":["arn:aws:s3:::${BUCKET_NAME}","arn:aws:s3:::${BUCKET_NAME}/*"]
+		},
+		{
+			"Effect":"Allow",
+			"Action":["ec2:ModifySnapshotAttribute","ec2:CopySnapshot","ec2:RegisterImage","ec2:Describe*"],
+			"Resource":"*"
+		}
+	]
+}
+EOF
+
+	aws iam put-role-policy --role-name vmimport --policy-name vmimport --policy-document "file://${ROLE_POLICY_JSON_FILE}" --region ${REGION}
+}
+
+clean_credentials() {
+	echo "Clean credentials"
+	oc delete credentialsrequest aws-vmimport -n openshift-cloud-credential-operator
+}
+
+get_credentials() {
+	[[ $skip_cr ]] && return
+	echo "Ask for addtional credentials"
+	oc get ns openshift-sandboxed-containers-operator >/dev/null 2>&1 || (echo "OSC namespace is missing, re-run after installting the operator" && exit 1)
+
+	cat <<EOF > "${CREDENTIALS_REQUEST_YAML_FILE}"
+apiVersion: cloudcredential.openshift.io/v1
```

Comment:
I noticed some variables here, like ${BUCKET_NAME}, and that might be why you're hardcoding it. But if it's not too much work, I suggest consolidating all requests under config/peerpods/credentials-requests. In this case, some substitution would be needed, but I still think having them there would help with our organization.

Reply:
Discussed offline, i set AWS credentials for image creation in separate until this will be the default method